torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -728,7 +728,8 @@ class TORCH_API ProcessGroupNCCL : public Backend {
       Fn fn,
       OpType opType,
       const char* profilingTitle = nullptr,
-      bool avoidRecordStreams = false);
+      bool avoidRecordStreams = false,
+      bool nanCheck = true);
```

Comment:
so we default to true which makes sense, bc most collectives (allreduce, allgather, gather, a2a) will need to check all inputs.  The ones that don't are special-cased (broadcast, scatter).  i think we covered them all.  
 
Is `_broadcast_oop` covered by broadcast?

Reply:
Thanks for checking. Added ` _broadcast_oop` now.